README.md

```diff
@@ -27,4 +27,120 @@ This is a repository for CPSC 410: Advanced Software Engineering. The TS-DSL sho
 
 ## Grammar
 
-TODO
+PROGRAM         ::= ‘project’ str nl (MODULE nl)? (DIR|CLASS|INTERFACE)*
+
+MODULE          ::= ‘modules’ STRLIST				# should have quotes
+
+DIR             ::= ‘dir’ str nl (ind (DIR|CLASS|INTERFACE) ded)* nl
+
+CLASS           ::= ABS? ‘class’ str IMP? EXT? (nl COMMENT)? (nl FIELD)* (nl FUNC)* nl
+
+INTERFACE       ::= ‘interface’ str EXT? (nl COMMENT)? (nl FIELD)* (nl FUNC)* nl
+
+FIELD           ::= ind ‘fields’ mod? VARLIST (nl ind 'generate' gen)* ded
+
+FUNC            ::= ind ‘function ’ mod ‘async ’? 'static'? str (nl ind COMMENT)? (nl PARAM)? (nl ind 'returns' type)? ded
+
+PARAM           ::= ind ‘params ’ VARLIST ded
+
+COMMENT         ::= ind ‘comments ’ STRLIST ded
+
+ABS             ::= ‘abstract’
+
+IMP             ::= ‘implements’ str (‘,’ str)*
+
+EXT             ::= ‘extends’ str
+
+VARLIST         ::= ‘[ ’ (type str (‘,’ type str))? ‘ ]’
+
+STRLIST         ::= ‘[ ‘ str (‘,’ str)* ‘ ]’
+
+gen             ::= (‘getters’ | ‘setters’)
+
+mod             ::= (‘private’ | ‘public’| ‘protected’)
+
+type            ::= (‘number’ | ‘boolean’| ‘string’ | str) sp
+
+str             ::= [^\n]* sp # note strings outside of comments cannot have spaces as well.
+
+sp              ::= ‘ ’
+
+nl              ::= [\n]+
+
+ind             ::= [\t | ]+    # tab(s) or single space(s)
+
+ded             ::= [] # this is a dedent
```

Comment:
Can you enclose this block in triple backticks? It'll look much more nicely formatted

Reply:
Good job @jyoo980, thanks!